### PR TITLE
fix: stylex example to use new vite plugin

### LINKED ISSUE #750: styles are not applied correctly with `stylex` in server components
Using stylex with https://github.com/nonzzz/vite-plugin-stylex not apply correctly the styles when the component is a server component. It is applied correctly when the component is a client component. I don't know where the error comes from, if it is from waku or from the plugin. If it is from the plugin then I would open a issue there.

I leave a repo to reproduce it: [https://stackblitz.com/edit/vitejs-vite-gj5y4r](https://stackblitz.com/edit/vitejs-vite-gj5y4r). 
The `Logotype` component should render with blue color but it is black when the server is running first time. If you save a random file like templates/root-layout.tsx then the `Logotype` component is correctly displays in blue. It happens only in dev mode, when the repository is built and started (`waku build`, `waku start`) everything works.

### PR DESCRIPTION
**Pull request by tylersayshi**
fixes #750

### CODE REVIEW

File: examples/37_css/vite.config.ts [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
 
 /** @type {import('vite').UserConfig} */
 export default {
   plugins: [
     {
-      name: 'hack-css-plugin-needed-for-stylex-dev-FIXME',
+      name: 'hack-css-plugin-needed-for-stylex-FIXME',
       resolveId(id: string) {
         if (id.endsWith('.css') && !id.endsWith('.vanilla.css')) {
           return id;
```
Comment (tylersayshi) [inline examples/37_css/vite.config.ts:11]: unfortunately it seems like the hack is still needed for now

**Review by himself65 — APPROVED** [review]
my friend in a telegram cn frontend group is maintaining `vite-plugin-stylex-dev`, I think he is not actively working on this now? @nonzzz 

He has no message since May 2024 🤔

**Review by dai-shi — APPROVED** [review]
Thanks for working on this!

**Comment by dai-shi** [comment]
There seems an issue with `pnpm -F 37_css dev`.

## first load
image

## after reload
image

Can you reproduce it on your end? @tylersayshi

**Comment by tylersayshi** [comment]
@dai-shi i see the same. I might not have time to debug tonight but could look closer soon.

This is specific to dev though. The issue is not present once we run waku build and waku start

**Comment by dai-shi** [comment]
No hurry. Yeah, dev only, but that's the issue. I think that's why we changed the plugin to `vite-plugin-stylex-dev`, or maybe.

**Pull request merged**